web/impact/impact/tests/test_user_confirm_participation_view.py

```diff
@@ -0,0 +1,150 @@
+        user = self.basic_user()
+        _get_programrolegrant_for_role(
+            UserRole.DEFERRED_MENTOR, user, self.program)
+        with self.login(email=user.email):
+            url = reverse(UserProgramConfirmationDetailView.view_name,
+                          args=[user.pk])
+            self.client.post(url, {
+                'confirmed': [self.program.pk],
+            })
+            response = self.client.get(url)
+            expected = {
+                'deferred': [],
+                'confirmed': [self.program.pk]
+            }
+            self.assertEqual(response.data['program_confirmation'], expected)
+
+    def test_deferring_program_removes_confirmed_status(self):
+        user = self.basic_user()
+        _get_programrolegrant_for_role(
+            UserRole.MENTOR, user, self.program)
+        with self.login(email=user.email):
+            url = reverse(UserProgramConfirmationDetailView.view_name,
+                          args=[user.pk])
+            self.client.post(url, {
+                'deferred': [self.program.pk],
+            })
+            response = self.client.get(url)
+            expected = {
+                'deferred': [self.program.pk],
+                'confirmed': []
+            }
+            self.assertEqual(response.data['program_confirmation'], expected)
+
+    def test_only_owner_can_access_participation_view(self):
+        non_owner_user = UserFactory()
+        with self.login(email=self.basic_user_without_api_groups().email):
+            url = reverse(UserProgramConfirmationDetailView.view_name,
+                          args=[non_owner_user.pk])
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, 403)
+
+    def test_cannot_post_invalid_input(self):
+        user = self.basic_user()
+        with self.login(email=user.email):
+            url = reverse(UserProgramConfirmationDetailView.view_name,
+                          args=[user.pk])
+            response = self.client.post(url, {
+                'deferred': ["test"],
+            })
+            expected = {
+                'detail': INVALID_INPUT_ERROR.format('deferred'),
+            }
+            self.assertEqual(response.data, expected)
+
+    def test_email_is_sent_after_confirming_participation(self):
+        user = self.basic_user()
+        with self.login(email=user.email):
+            url = reverse(UserProgramConfirmationDetailView.view_name,
+                          args=[user.pk])
+            self.client.post(url, {
+                'confirmed': [self.program.pk],
+            })
+        self.assertEqual(mail.outbox[0].subject, SUBJECT)
+
+
+def _get_user_programs_for_role(user, role):
+    return list(
+        user.programrolegrant_set.filter(
+            program_role__user_role__name=role,
+        ).values_list('program_role__program', flat=True).distinct())
+
+
+def _get_programrolegrant_for_role(role, user, program=None):
```

Comment:
since this creates rather than getting, I would like to call it `_create_program_role_grant_for_role` (use underscores to separate words!)